### PR TITLE
create bin directory if not present

### LINKED ISSUE #159: [CI] kubeopenapi-jsonschdema is not downloading when running `make run-fast`
#### Current Behavior
When I run the command `make run-fast` I get the following error related to `kubeopenapi-jsonschema` 
```
Looking for kubeopenapi-jsonschema in  /Users/aaditya/.meshery/bin
Downloading kubeopenapi-jsonschema at /Users/aaditya/.meshery/bin/kubeopenapi-jsonschema-darwin...
ERRO[2022-03-03T20:30:58+05:30] open /Users/aaditya/.meshery/bin/kubeopenapi-jsonschema-darwin: no such file or directory  app=meshery code=2177 probable-cause="component generation was canceled due to deletion or reload of K8s context.Invalid kubeconfig.Filters passed incorrectly in config.Could not fetch API resources from Kubernetes server" severity=2 short-description="failed to register/create kubernetes components for contextID 5258c1b748fb4f056fb0c72ac2c4b177" suggested-remediation="If there is the log \"Starting to register ...\" for the same contextID after this error means that for some reason the context was reloaded which caused this run to abort. In that case, this error can be ignored..Make sure that the configuration filters passed are in accordance with output from /openapi/v2"

```
I looked into the `~/.meshery` folder and could not find any `bin` folder created.
#### Acceptance Tests

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #159 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Revolyssup** [comment]
@asubedy can you checkout this PR and test if the bin directory gets created now for you?

steps:
1. Checkout to this pr.
2. add the below line in your go.mod of istio adapter and then run go mod tidy. Then run FORCE_DYNAMIC_REG=true DEBUG=true go run main.go
`replace github.com/layer5io/meshkit <version> => "<path to your local meshkit>"`

File: utils/manifests/generateComponent.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ func GenerateComponents(ctx context.Context, manifest string, resource int, cfg
 		inputFormat = "json"
 	}
 	wd := filepath.Join(utils.GetHome(), ".meshery", "bin")
+	err := os.Mkdir(wd, 0750)
+	if err != nil && !os.IsExist(err) {
+		return nil, err
+	}
 	fmt.Println("Looking for kubeopenapi-jsonschema in ", wd)
```
Comment (leecalcote) [inline utils/manifests/generateComponent.go:22]: Should this be our MeshKit logger?

File: utils/manifests/generateComponent.go [hunk 2]
```diff
@@ -19,6 +19,10 @@ func GenerateComponents(ctx context.Context, manifest string, resource int, cfg
 		inputFormat = "json"
 	}
 	wd := filepath.Join(utils.GetHome(), ".meshery", "bin")
+	err := os.Mkdir(wd, 0750)
+	if err != nil && !os.IsExist(err) {
+		return nil, err
```
Comment (leecalcote) [inline utils/manifests/generateComponent.go:24]: Where do we have a MeshKit error with suggestion to verify file system permissions on failure?

**Comment by leecalcote** [comment]
When leaving my first comment, I didn't realize that this PR is in MeshKit not in Meshery Server.

File: utils/manifests/generateComponent.go (hunk 2 above)
Comment (leecalcote) [reply utils/manifests/generateComponent.go:24]: Details for the user on what do do in this case of this error would be good.

Comment (leecalcote) [reply utils/manifests/generateComponent.go:24]: Reminder...

Comment (Revolyssup) [reply utils/manifests/generateComponent.go]: done

**Review by leecalcote — APPROVED** [review]
Looks great. 👍

**Pull request merged**